app/controllers/games_controller.rb

```diff
@@ -9,15 +9,21 @@ def index
     else
       @games = Game.top_five
     end
-    render json: @games.as_json(:include => {:sessions => {only: [:id, :date, :victor]}})
+
+    include_options = build_include_options
+    render json: @games.as_json(include: include_options)
   end
 
   # GET /games/1 or /games/1.json
   def show
+
+    include_options = build_include_options
+
     @game = Game.find(params[:id])
     if @game 
-      @result = JSON.parse(@game.game_relationships)
-      @result['results'] = @game.results
+      #@result = JSON.parse(@game.game_relationships)
+      @result = @game.as_json(include: include_options)
+      @result['results'] = @game.results if params[:include]&.split(',')&.include?('results')
```

Comment:
thanks for these tips! I can definitely make the variable name more clear. And I will read up on serializer; I know next to nothing about them, so this will be a good tool to add to my arsenal!